### PR TITLE
feat: `pm-x` generator

### LINKED ISSUE #5: package manager command generators
- [x] `pm-install` (#9)
- [ ] `pm-x`

### PR DESCRIPTION
**Pull request by uncenter**
> Oops, looks like renaming the branch causes the PR to be closed... live and learn I guess.

### 🔗 Linked issue

Closes #5.

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme, or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Adds a `pm-x` generator as discussed in #5. Implementation based on `pm-install` generator.

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by cpreston321** [comment]
@uncenter update main README.md 🙏🏼?

**Comment by uncenter** [comment]
> @uncenter update main README.md 🙏🏼?

Right yeah I forgot ahah. Should be good now.

File: src/generators/pm/utils.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { readPackageJSON } from "pkg-types";
+
+export async function inferPackageName(dir: string) {
+  const pkgName = await readPackageJSON(dir)
+    .then((pkg) => pkg?.name)
+    .catch(() => undefined);
+  return pkgName || process.env.npm_package_name;
+}
```
Comment (cpreston321) [inline src/generators/pm/utils.ts:8]: I was going to use this util for `badges` maybe it util root vs `pm/util.ts`

**Review by pi0 — APPROVED** [review]
Thanks! (added some refactors and also ideas mainly to try!)

File: src/generators/pm/utils.ts (hunk 1 above)
Comment (pi0) [reply src/generators/pm/utils.ts]: refactored 👍🏼

**Comment by uncenter** [comment]
Sweet, changes looks good!

**Pull request merged**